### PR TITLE
Add devise views to project

### PR DESCRIPTION
**Pull request by asolvang**
This PR has some work left. 
A couple of the Devise views are left, will be completed tomorrow. (new password and password reset)
Login and signup works.

Also added flash messages.

To test: 
yarn
then run project as normally.

### CODE REVIEW

**Comment by asolvang** [comment]
[Frontend: Add Devise views](https://trello.com/c/i7HVxKiU/48-frontend-add-devise-views)

**Comment by DanielJackson-Oslo** [comment]
WIP, obviously, but checked it out now and looks very cool! Looking forward to getting this merged :) 

We have to decide on colours at some point. Currently I've been using a non-LNU pink all over, and it would make sense to standardize on the LNU one - but I haven't prioritzed it yet. I'm thinking I will change the pink-XXX in tailwind for custom pinks generated from LNU pink at some point.

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <%= f.password_field :current_password, autocomplete: "current-password", class: 'passwordField z-0', placeholder: '******', minlength: '6', required: true %>
+      </div>
+    </div>
+    <div class="mt-4">
+      <%= button_tag :submit, class: 'group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-lnu-pink hover:bg-lnu-blue focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-lnu-pink' do %>
+        <span class="absolute left-0 inset-y-0 flex items-center pl-3">
+          <!-- Heroicon name: solid/lock-closed -->
+          <svg class="h-5 w-5 text-white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+            <path fill-rule="evenodd" d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z" clip-rule="evenodd" />
+          </svg>
```
Comment (mathias234) [inline app/views/devise/registrations/edit.html.erb:42]: See other comment

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          <%= f.password_field :password_confirmation, autocomplete: "new-password", class: 'passwordField fieldLast z-10', placeholder: '******', required: true %>
+        </div>
+      </div>
+      <div class="mt-4">
+        <%= button_tag :submit, class: 'group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-lnu-pink hover:bg-lnu-blue focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-lnu-pink' do %>
+          <span class="absolute left-0 inset-y-0 flex items-center pl-3">
+            <!-- Heroicon name: solid/lock-closed -->
+            <svg class="h-5 w-5 text-white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+              <path fill-rule="evenodd" d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z" clip-rule="evenodd" />
+            </svg>
```
Comment (mathias234) [inline app/views/devise/registrations/new.html.erb:37]: See other comment

File: app/views/shared/_notice.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="flex-shrink-0">
+  <svg class="h-6 w-6 text-green-400" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
+    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
+  </svg>
```
Comment (mathias234) [inline app/views/shared/_notice.html.erb:4]: See other comment

**Review by mathias234 — CHANGES_REQUESTED** [review]
Looks good, just "one" comment about using inline_svg as we use that for all our other icons

**Comment by asolvang** [comment]
@DanielJackson-Oslo 
> We have to decide on colours at some point. Currently I've...

I setup corlor LNU-> Pink/Blue/Green
Took them from the Logo.

**Review by mathias234 — APPROVED** [review]
LGTM!

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -12,12 +12,13 @@
...
   <body>
-    <nav class="grid grid-cols-12 mt-2">
     <% if current_user %>
+      <nav class="grid grid-cols-12 mt-2">
         <%= link_to "Signout", destroy_user_session_path, class: "col-start-11 col-span-1 p-1 font-l text-center border-4 border-green-400 bg-green-100 rounded-md", method: :delete %>
+      </nav>
     <% end %>
-    </nav>
-    <main>
+    <main data-controller="example" class="min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8">
```
Comment (DanielJackson-Oslo) [inline app/views/layouts/application.html.erb:20]: data-controller="example" seems a bit weird! Left over from some other code?

File: app/views/devise/registrations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+  <div class="max-w-md w-full space-y-8">
+    <div>
+      <%= image_tag 'logo.svg', alt: 'LNU logo', width: 200, height: 200, class: 'mx-auto h-12 w-auto' %>
```
Comment (DanielJackson-Oslo) [inline app/views/devise/registrations/new.html.erb:3]: This code is duplicated between views registrations/new and sessions/new. Should be pulled out to it's own little thing?

**Review by DanielJackson-Oslo — COMMENTED** [review]
Some comments for posterity and future PRs!

**Pull request merged**